### PR TITLE
[AJ-1766] Convert ModalDrawer to TypeScript

### PR DESCRIPTION
**Pull request by nawatts**
Continuing to work on converting AzureComputeModal to TypeScript.

This converts ModalDrawer to TypeScript. In order to satisfy TypeScript, I had to convert to TSX (our types for React Hyperscript Helpers have some issues) and slightly change `withModalDrawer`'s behavior to pass through more props. This shouldn't have any effect though. The case where it could is explained in a comment.

Suggest reviewing one commit at a time to see more understandable diffs.

### CODE REVIEW

**Review by jdcanas — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -102,6 +102,8 @@
     "@types/node": "^18.7.16",
     "@types/react": "^18.2.15",
     "@types/react-dom": "^18.2.7",
+    "@types/react-modal": "^3.16.2",
```
Comment (cahrens) [inline package.json:105]: Why do we need these in dev dependencies?

Comment (nawatts) [reply package.json:105]: These packages contain type definitions for plain JS packages that don't provide type definitions themselves.
https://definitelytyped.org/

TypeScript automatically uses them.
https://www.typescriptlang.org/tsconfig/#typeRoots

File: src/components/ModalDrawer.tsx [hunk 2]
```diff
@@ -79,12 +79,14 @@ const ModalDrawer = (props: ModalDrawerProps): ReactNode => {
...
-      const { isOpen, onDismiss, onExited, ...otherProps } = props;
+  <P extends { onDismiss: () => void }>(
+    WrappedComponent: ComponentType<P>
+  ): FunctionComponent<P & Pick<ModalDrawerProps, 'isOpen' | 'onExited'>> => {
+    const Wrapper = (props: P & Pick<ModalDrawerProps, 'isOpen' | 'onExited'>): ReactNode => {
+      const { isOpen, onDismiss, onExited } = props;
       return (
         <ModalDrawer isOpen={isOpen} onDismiss={onDismiss} onExited={onExited} {...modalDrawerProps}>
-          {isOpen && <WrappedComponent {...{ onDismiss, ...otherProps }} />}
+          {isOpen && <WrappedComponent {...props} />}
```
Comment (cahrens) [inline src/components/ModalDrawer.tsx:115]: Isn't this a change in behavior in that now all the props are being passed to `WrappedComponent`?

Comment (nawatts) [reply src/components/ModalDrawer.tsx:115]: Yes, it is. I had to do this to satisfy TypeScript. The alternative was casting `otherProps` to `any`. The comment in the last commit explains why I believe this change won't have any effect. Everything currently wrapped in `withModalDrawer` ignores the extra props.

**Review by cahrens — APPROVED** [review]

**Pull request merged**